interpreter/utils/edit_code.py

```diff
@@ -0,0 +1,64 @@
+import os
+from yaspin import yaspin
+
+from .temporary_file import create_temporary_file, cleanup_temporary_file
+from .open_file import open_file
+from ..code_interpreters.languages.utils.language_tools import get_language_file_extension, get_language_proper_name
+
+
+def edit_code(code, language, interpreter):
+    """
+    Edit the code and listen for changes with watchdog
+    """
+
+    temp_code = code
+
+    temp_file = create_temporary_file(
+        temp_code, get_language_file_extension(language), verbose=interpreter.debug_mode
+    )
+
+    language_name = get_language_proper_name(language)
+
+    file_name = os.path.basename(temp_file)
+
+    if interpreter.debug_mode:
+        print(f"Editing {language_name} code in {file_name}")
+        print("---")
+
+    # Run semgrep
+    try:
+        print("  Press `ENTER` after you've saved your edits.")
+
+        open_file(temp_file)
+
+        with yaspin(text=f"  Editing {language_name} code...").green.right.dots as loading:
```

Comment:
There's probably a better way to handle this. The big problem right now is that since we don't really get any kind of useful signal when the user is done, we can't automatically clean up the spinner like we do with code scanning.

This probably needs to shift to an imperative call to a new `yaspin` instance instead of using `with yaspin()` so we have more control over the loading indicator.